pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -21,25 +22,23 @@ def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_a
         self.randomise_all_colors = randomise_all_colors
         self.shuffle_obs_per_agent = shuffle_obs_per_agent
 
-        if num_colors == None:
-            self.num_colors = len(self.groups)
-        else:
-            self.num_colors = num_colors
+        self.num_colors = len(self.groups) if num_colors is None else num_colors
+        self.obs_onehot_colors = obs_onehot_colors
 
         if not self.randomise_all_colors:
```

Comment:
Previously colors are randomly generated with `np.random.random(3)` but this can lead to very close/ overlapping colors. I agree this is unlikely but evenly spaced out colors might be easier to distinguish programmatically.